### ISSUE TITLE
Search *

### ISSUE THREAD

**Issue by widal001** [HHS/simpler-grants-gov#2203]
### Summary

Grant seekers can search, filter, and sort opportunities on simpler.grants.gov

### Acceptance criteria

- [ ] **Search, filter, and sort:** Users can
  - [ ] search opportunities by keyword
  - [ ] filter by at least 3 structured fields
  - [ ] sort by relevance and 3 additional fields
- [ ] **Stateful UI:** Search criteria are reflected in the URL, so that users can bookmark or share the link to a page with a given combination of search criteria.
- [ ] **API parity:** All search features available via the UI are also available via API.

### Metrics

- [ ] **Adoption:** At least 10,000 unique users have used the search UI, and at least 5 unique users have used the search directly from the API endpoint (i.e., not via the UI).
- [ ] **Reliability:** The monthly average error rate remains below 5%, meaning fewer than 5 failures per 100 API and page requests in production.
- [ ] **Search accuracy:** There is a 10% increase in the monthly click-through rate (CTR) for results on the first page.

**Comment by mxk0** [comment]
After reviewing with the team, we should change "Download NOFO package" to "Download NOFO document". (Could be PDF, Word, .txt, etc.)

**Comment by mdragon** [comment]
> After reviewing with the team, we should change "Download NOFO package" to "Download NOFO document". (Could be PDF, Word, .txt, etc.)

Made those edits with Strikethrough, but @widal001 we can discuss more if we're not ok with the way it reads now.

**Comment by widal001** [comment]
> > After reviewing with the team, we should change "Download NOFO package" to "Download NOFO document". (Could be PDF, Word, .txt, etc.)
> 
> Made those edits with Strikethrough, but [@widal001](https://github.com/widal001) we can discuss more if we're not ok with the way it reads now.

Awesome thanks @mdragon! Thanks for looking into this further and updating the language to reflect the distinction between packages and documents and the support for file formats other than PDF.

**Comment by margaretspring** [comment]
@widal001 - Confirming my team is all good with updates to send the commitments to Julius for approval. Ok if we go ahead & incorporate the suggested update Matt provided?

**Comment by widal001** [comment]
@margaretspring Yes! Feel free to drop the text with the strike through.

**Comment by mxk0** [comment]
## Sprint 1.2 updates

**Status:** 🟢 On track

### Sprint goal

The main goal for this deliverable in Sprint 1.2 was to develop and ADR for NOFO storage and make the needed backend changes.

### Accomplishments

- Added a way for user to learn more or apply for an opportunity (via Grants.gov)
- Improvements to the UI
- Fixed inconsistencies when there are null values

### Rollover

- N/A

### Risks

- We don't see any large risks at this time.

### Sprint 1.3 goals

- Publicly launch Opportunity Listing!

**Comment by lucasmbrown-usds** [comment]
"The main goal for this deliverable in Sprint 1.2 was to develop an ADR for NOFO storage and make the needed backend changes." - did we end up accomplishing that?

**Comment by mdragon** [comment]
> "The main goal for this deliverable in Sprint 1.2 was to develop an ADR for NOFO storage and make the needed backend changes." - did we end up accomplishing that?

Yup, we have [the ADR](https://github.com/HHS/simpler-grants-gov/blob/main/documentation/wiki/decisions/adr/2024-10-18-document-storage.md) and the implementation isn't in Prod yet, but that's what Michael demoed (after you asked this question 😃)

**Comment by mdragon** [comment]
## Sprint 1.3 updates

**Status:** 🟢 On track

### Sprint goal

The main goal for this deliverable in Sprint 1.3 was to launch the Opportunities page publically

### Accomplishments
- Publicly launched the new page, removed the feature flag

### Rollover
- Expose Documents and NOFOs for our Pilot Opportunities in the UI

### Risks
- Traffic goal for Opportunity page is lower, but this is downstream of Search traffic that we're struggling to drive to the new site so hitting the mark is at risk.

### Sprint 1.4 goals
- Finish NOFO/Document UI work
- Fix bugs/issues
- Identify iterative changes driven by UX Research

**Comment by lucasmbrown-usds** [comment]
I'm very excited for 1.4 to finish the NOFO/Document UI work!

**Comment by mxk0** [comment]
## Sprint 1.4 updates

**Status:** 🟢 On track

### Sprint goal

The main goal for this deliverable in Sprint 1.4 was to deliver NOFO attachment functionality in prod.

### Accomplishments

- Delivered our 7 hand-picked NOFOs in prod (e.g. https://simpler.grants.gov/opportunity/349135)
- Continued driving traffic to Search/Opp Listing

### Rollover
- 
- N/A

### Risks

- User traffic numbers are still at risk

### Sprint 1.5 goals

- Continue driving traffic

**Comment by mdragon** [comment]
![Image](https://github.com/user-attachments/assets/58470b8b-e4a5-47c4-825e-1b29c29898ea)
Data is syncing hourly. On 30 minute offsets we pull data from the grants.gov side and then transform it to meet our data model and push it into search. So this shows 48 runs in the past day, 24 of the load, 24 of the transform, occurring every 30 minutes alternating tasks. [Source](https://us-east-1.console.aws.amazon.com/cloudwatch/home?region=us-east-1#logsV2:logs-insights$3FqueryDetail$3D~(end~0~start~-86400~timeType~'RELATIVE~tz~'UTC~unit~'seconds~editorString~'filter*20funcName*20*3d*20*27_ecs_background_task_impl*27*20*7c*20filter*20not*20isempty*28status*29*0a*7c*20fields*20*20*40timestamp*2c*20task_name*2c*20status*2c*20msg*2c*20*40message~queryId~'5ade2b73-a307-42f9-90a0-3fa587d9ccd5~source~(~'service*2fapi-prod)~lang~'CWLI))

**Comment by mxk0** [comment]
# DRAFT Enhanced Deliverable Wrap-Up

## Acceptance Criteria

- [ ] Listing page: Deliver a working opportunity listing page that allows users to learn about an opportunity (at least title and description) on simpler.grants.gov.

- [ ] Download NOFO: For at least 5 SimplerNOFOs, and at least 2 other non-PDF file types, the opportunity listing page supports downloading the NOFO (PDF, DOCX, TXT) in just one click.

- [ ] Learn more link: The opportunity listing page contains a “Learn more and apply” link that directs user to the corresponding opportunity listing page on grants.gov.

- [ ] Load testing: Load testing performed to assess whether we can simulate 500 concurrent users with our current infrastructure.

## Metrics

- [ ] UI adoption: At least 2,000 unique users have viewed the opportunity listing via the user interface (UI).

- [ ] Database syncing: Set a baseline metric for how quickly data is synced from Grants.gov to simpler.grants.gov; i.e., data is synced at least once every 24 hours.

- [ ] Conversion: Set a baseline metric for percent of users who click “Learn more and apply” on the opportunity listing page.
